x-pack/plugins/security/server/lib/saved_objects_client/secure_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,121 @@
+  errors: any;
+}
+export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContract {
+  private baseClient: SavedObjectsClientContract;
+
+  private checkSavedObjectsPrivileges: CheckSavedObjectsPrivileges;
+
+  public errors: any;
+
+  constructor(options: SecureSavedObjectsClientWrapperDeps) {
+    const { baseClient, checkSavedObjectsPrivileges, errors } = options;
+
+    this.errors = errors;
+    this.baseClient = baseClient;
+    this.checkSavedObjectsPrivileges = checkSavedObjectsPrivileges;
+  }
+
+  public async create<T extends SavedObjectAttributes = any>(
+    type: string,
+    attributes: T,
+    options: CreateOptions = {}
+  ) {
+    await this.ensureAuthorized(type, 'create', options.namespace, { type, attributes, options });
+
+    return await this.baseClient.create(type, attributes, options);
+  }
+
+  public async bulkCreate<T extends SavedObjectAttributes = any>(
+    objects: Array<BulkCreateObject<T>>,
+    options: CreateOptions = {}
+  ) {
+    const types = uniq(objects.map(o => o.type));
+    await this.ensureAuthorized(types, 'bulk_create', options.namespace, { objects, options });
+
+    return await this.baseClient.bulkCreate(objects, options);
+  }
+
+  public async delete(type: string, id: string, options: BaseOptions = {}) {
+    await this.ensureAuthorized(type, 'delete', options.namespace, { type, id, options });
+
+    return await this.baseClient.delete(type, id, options);
+  }
+
+  public async find<T extends SavedObjectAttributes = any>(
+    options: FindOptions = {}
+  ): Promise<FindResponse<T>> {
+    await this.ensureAuthorized(options.type, 'find', options.namespace, { options });
+
+    return this.baseClient.find(options);
+  }
+
+  public async bulkGet<T extends SavedObjectAttributes = any>(
+    objects: BulkGetObject[] = [],
+    options: BaseOptions = {}
+  ): Promise<BulkResponse<T>> {
+    const types = uniq(objects.map(o => o.type));
+    await this.ensureAuthorized(types, 'bulk_get', options.namespace, { objects, options });
+
+    return await this.baseClient.bulkGet(objects, options);
+  }
+
+  public async get<T extends SavedObjectAttributes = any>(
+    type: string,
+    id: string,
+    options: BaseOptions = {}
+  ): Promise<SavedObject<T>> {
+    await this.ensureAuthorized(type, 'get', options.namespace, { type, id, options });
+
+    return await this.baseClient.get(type, id, options);
+  }
+
+  public async update<T extends SavedObjectAttributes = any>(
+    type: string,
+    id: string,
+    attributes: Partial<T>,
+    options: UpdateOptions = {}
+  ) {
+    await this.ensureAuthorized(type, 'update', options.namespace, {
+      type,
+      id,
+      attributes,
+      options,
+    });
+
+    return await this.baseClient.update(type, id, attributes, options);
+  }
+
+  private async ensureAuthorized(
+    typeOrTypes: string | string[] | undefined,
+    action: SavedObjectsOperation,
+    namespace: string | undefined,
```

Comment:
The diff isn't very helpful since this was converted to TypeScript, but the only material difference in this file is the introduction of this `namespace` parameter, and the swapping of `checkPrivilegesDynamicallyWithRequest` with `checkSavedObjectsPrivileges`.